src/AutoRest.CSharp/MgmtTest/Generation/Samples/MgmtSampleWriter.cs

```diff
@@ -362,27 +384,46 @@ private CodeWriterVariableDeclaration WriteGetCollection(CodeWriterVariableDecla
             var returnType = sample.Operation.ReturnType;
             if (returnType.IsGenericType)
             {
-                // an operation with a response
-                var unwrappedReturnType = returnType.Arguments.First();
-                if (unwrappedReturnType.IsGenericType) // if the type inside Response<T> is a generic type, somehow the implicit convert Response<T> => T does not work, we have to explicitly unwrap it
+                // if the type is NullableResponse, there is no implicit convert, so have to explicitly unwrap it
+                if (returnType.Name == nameof(NullableResponse<string>))
```

Comment:
Could we do something like this instead of asserting this by its name?
```suggestion
                if (returnType.IsFrameworkType && returnType.FrameworkType == typeof(NullableResponse<>))
```